app/main.py

```diff
@@ -391,24 +402,27 @@ def get_discover_path():
 # important to keep the relative <path> for accessing
 # other required files.
 @app.route("/s3-resource/<path:path>")
-def direct_download_url(path):
+def direct_download_url(path, bucket_name=Config.S3_BUCKET_NAME):
+
+    query_args = request.args
+    s3BucketName = query_args.get("s3BucketName") or bucket_name
+
     try:
         head_response = s3.head_object(
-            Bucket=Config.S3_BUCKET_NAME,
+            Bucket=s3BucketName,
             Key=path,
             RequestPayer="requester"
         )
+        content_length = head_response.get('ContentLength', Config.DIRECT_DOWNLOAD_LIMIT)
+        if content_length and content_length > Config.DIRECT_DOWNLOAD_LIMIT:  # 20 MB
```

Comment:
Surely `content_length` is always defined here? If the `ContentLength` header is not set then you are happy to download a file of any size?

Reply:
Hmmm, I moved this block into the try block without looking into the logic but you are right, I will change that.